be/extmap.c

```diff
@@ -917,37 +1053,60 @@ static int emap_it_open(struct m0_be_emap_cursor *it)
 	rc = op->bo_u.u_btree.t_rc;
 	if (rc == 0) {
 		m0_be_btree_cursor_kv_get(&it->ec_cursor, &keybuf, &recbuf);
+
+		/* Key operation */
 		key = keybuf.b_addr;
-		rec = recbuf.b_addr;
 		it->ec_key = *key;
+
+		/* Record operation */
+		if (it->ec_recbuf.b_addr != NULL) {
+			m0_buf_free(&it->ec_recbuf);
+		}
+
+		/* Layout/format of emap-record (if checksum is present) which
+		 * gets written:
+		 * -[Hdr|Balloc-Ext-Start|B-Ext-Value|CS-nob|CS-Array[...]| Ftr]
+		 * It gets stored as contigious buffer, so allocating buffer
+		 */
+		rc = m0_buf_alloc(&it->ec_recbuf, recbuf.b_nob);
+		if ( rc != 0)
+			return rc;
+
+		/* Copying record buffer and loading into it->ec_rec, note
+		 * record will have incorrect footer in case of b_nob,
```

Comment:
[* style *] use tab.
same for next line.

Reply:
Done.